send.php

```diff
@@ -29,64 +29,73 @@
 
 $dir = dirname($file);
 $old = $file;
-$file = $dir . '/analytics-' . rand() . '.log';
 
-if(!file_exists($old)) {
-  print("file: $old does not exist");
-  exit(0);
-}
+$newFileName = 'analytics-' . rand() . '.log';
+$newFile = $dir . '/' . $newFileName;
 
-if (!rename($old, $file)) {
-  print("error renaming from $old to $file\n");
-  exit(1);
+if(file_exists($old)) {
+    if (!rename($old, $newFile)) {
+        print("error renaming from $old to $newFile\n");
+        exit(1);
+    }
 }
 
-/**
- * File contents.
- */
-
-$contents = file_get_contents($file);
-$lines = explode("\n", $contents);
-
-/**
- * Initialize the client.
- */
-
-Segment::init($args["secret"], array(
-  "debug" => true,
-  "error_handler" => function($code, $msg){
-    print("$code: $msg\n");
-    exit(1);
-  }
-));
-
-/**
- * Payloads
- */
-
-$total = 0;
-$successful = 0;
-foreach ($lines as $line) {
-  if (!trim($line)) continue;
-  $payload = json_decode($line, true);
-  $dt = new DateTime($payload["timestamp"]);
-  $ts = floatval($dt->getTimestamp() . "." . $dt->format("u"));
-  $payload["timestamp"] = $ts;
-  $type = $payload["type"];
-  $ret = call_user_func_array(array("Segment", $type), array($payload));
-  if ($ret) $successful++;
-  $total++;
-  if ($total % 100 === 0) Segment::flush();
+foreach(scandir($dir) as $file) {
+    if (substr($file, 0, strlen('analytics-')) !== 'analytics-') {
```

Comment:
1. This feels kinda fragile.
A. What if there's a lot unrelated files starting with `analytics-`?
B. How do we know that those files start with `analytics-` in the first place?
C. If a cron run takes more than a minute and another one starts up in parallel, you could probably end up with duplicate events.
2. You could also write this as `strpos($file, 'analytics-') !== 0`

Reply:
1B. because we rename them to that :)
1A. that's fair I suppose, but I don't really see a solution to this?
1C. Very good catch! I guess we can have a lockfile (`analytics.lock`?) in the same directory to prevent two crons running at the same time.
2. Yes.

Reply:
1B. My bad, didn't really get how it worked.
1A. Maybe we could try using a more unique name? Like a hash of the current directory path or something like that.
1C. Yeah, or we could also use [flock](https://linux.die.net/man/1/flock)